modules/basicfuncs/tf-map.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+typedef struct _MapState
+{
+  TFSimpleFuncState super;
+  LogTemplate *template;
+} MapState;
+
+static gboolean
+tf_map_prepare(LogTemplateFunction *self, gpointer s, LogTemplate *parent, gint argc, gchar *argv[],
+               GError **error)
+{
+  MapState *state = (MapState *)s;
+
+  GOptionContext *ctx = g_option_context_new("map");
```

Comment:
What is `ctx` used for?

Reply:
I did not want to deal with shell escaping by hand, that's why I used argparser from glib. That needs a context to work. I just tried, and `g_option_context_parse` with `NULL` causes crash.

Reply:
Hmm I confused it with something. You are right. Context is not needed here.